### PR TITLE
planning: open space smoother code clean

### PR DESCRIPTION
**Pull request by deidaraho**
1, print out ipopt failure code
2, remove by hand gradient calculation function and proto option
3, remove cpu-based parallel computing Jacobian function and proto option

### CODE REVIEW

File: modules/planning/open_space/trajectory_smoother/distance_approach_problem.cc [hunk 1]
```diff
@@ -139,7 +140,34 @@ bool DistanceApproachProblem::Solve(
...
+        "Not_Enough_Degrees_Of_Freedom",
+        "Invalid_Problem_Definition",
+        "Invalid_Option",
+        "Invalid_Number_Detected",
+        "Unrecoverable_Exception",
+        "NonIpopt_Exception_Thrown"
+        "Insufficient_Memory",
+        "Internal_Error"};
+    if (static_cast<size_t>(status) >= failure_status.size()) {
+      AINFO << "Solver ends with unknow falure code: "
```
Comment (xiaoxq) [inline modules/planning/open_space/trajectory_smoother/distance_approach_problem.cc]: Multi typos.

Comment (deidaraho) [reply modules/planning/open_space/trajectory_smoother/distance_approach_problem.cc]: > Multi typos.

I find two, one in line 144 and the other in line 165. Are there some more?

Comment (xiaoxq) [reply modules/planning/open_space/trajectory_smoother/distance_approach_problem.cc]: unknown.

Comment (deidaraho) [reply modules/planning/open_space/trajectory_smoother/distance_approach_problem.cc]: > unknown.

THX, just fixed

**Pull request merged**